dot/sync/chain_sync.go

```diff
@@ -525,11 +522,15 @@ func (cs *chainSync) getTarget() *big.Int {
 		return big.NewInt(2<<32 - 1)
 	}
 
+	// we are going to sort the data and remove the outliers then we will return the avg of all the valid elements
+	intArr := make([]*big.Int, len(cs.peerState))
+	i := 0
 	for _, ps := range cs.peerState {
-		sum = big.NewInt(0).Add(sum, ps.number)
-		count++
+		intArr[i] = ps.number
+		i++
```

Comment:
nit you could init the array with `intArr := make([]*big.Int, 0, len(cs.peerState))` and use `append` so that you don't need to define and use `i` at all. It's a bit cleaner imo when iterating over a map and placing things in a slice.

Reply:
Duplicate suggestion of https://github.com/ChainSafe/gossamer/pull/1992#discussion_r749446999